### ISSUE TITLE
Neo.DatabaseError.General.UnknownError when using property replacement/mutation operator

### ISSUE THREAD

**Issue by hs750** [neo4j/neo4j#12249]
I have a database that is currently populated with around 2000 nodes. I am trying to insert a data that was exported from APOC as plain cypher form another instance of the database. 

The following statement is the one that fails:
`UNWIND [{_id:48, properties:{password:"asdf", authToken:"6a4e0954-3d78-4601-bd7f-0825f9838841", screenName:"asdf", uuid:"96c5e031-8b6c-11e9-9c7e-f8b156dff1f6", username:"asdf"}}] as row
CREATE (n:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`: row._id}) SET n += row.properties SET n:User;`

Removing `n+=row.properties` allows the command to run. Changing `+=` to `=` still fails. 

Not all usages of `+=` cause a failure. `match(n) set n += {test: "hello"}` works. As do some other commands in the exported cypher such as: 
`UNWIND [{_id:1290, properties:{default:true, name:"default", uuid:"96c5e030-8b6c-11e9-9c7e-f8b156dff1f6"}}, {_id:1414, properties:{name:"demo", uuid:"fd460d90-a4c5-11e9-9266-f7d63871dc56"}}] as row
CREATE (n:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`: row._id}) SET n += row.properties SET n:Context;`

This issue also does not happen if the database is empty.

The first command before inport is `CREATE CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT (node.`UNIQUE IMPORT ID`) IS UNIQUE;` and the last is `DROP CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT (node.`UNIQUE IMPORT ID`) IS UNIQUE;`. These commands have no effect on whether the other statements work. 

**Neo4j Version:** 3.5.5+ (works ok in 3.5.4)
**Operating System:** Windows 7 / Ubuntu 18.04 + Docker (alpine linux)
**API:** JavaScript Driver + Neo4j Browser
**Plugins:** Error happens without plugins. The following were installed before debugging: Graphaware Server, Graphaware UUID, APOC.

### Expected behavior
Data imported into neo4j.

### Actual behavior
Message in neo4j browser of `Neo.DatabaseError.General.UnknownError`.
Exception in neo4j log of:
```
2019-07-19 08:21:24.329+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: null, reference d9885c32-b552-4427-b69f-b1b70b37a245.
java.lang.NullPointerException
	at org.neo4j.values.storable.ValueTuple.hashCode(ValueTuple.java:108)
	at java.util.HashMap.hash(Unknown Source)
	at java.util.HashMap.computeIfAbsent(Unknown Source)
	at org.neo4j.kernel.impl.api.state.TxState.getOrCreateIndexUpdatesForSeek(TxState.java:803)
	at org.neo4j.kernel.impl.api.state.TxState.indexDoUpdateEntry(TxState.java:786)
	at org.neo4j.kernel.impl.newapi.IndexTxStateUpdater.onLabelChange(IndexTxStateUpdater.java:87)
	at org.neo4j.kernel.impl.newapi.Operations.checkConstraintsAndAddLabelToNode(Operations.java:303)
	at org.neo4j.kernel.impl.newapi.Operations.nodeAddLabel(Operations.java:273)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext$$anonfun$setLabelsOnNode$1.apply$mcIII$sp(TransactionBoundQueryContext.scala:81)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext$$anonfun$setLabelsOnNode$1.apply(TransactionBoundQueryContext.scala:80)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext$$anonfun$setLabelsOnNode$1.apply(TransactionBoundQueryContext.scala:80)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1334)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext.setLabelsOnNode(TransactionBoundQueryContext.scala:80)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$setLabelsOnNode$1.apply$mcI$sp(ExceptionTranslatingQueryContext.scala:53)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$setLabelsOnNode$1.apply(ExceptionTranslatingQueryContext.scala:53)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$setLabelsOnNode$1.apply(ExceptionTranslatingQueryContext.scala:53)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$class.translateException(ExceptionTranslationSupport.scala:33)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.translateException(ExceptionTranslatingQueryContext.scala:41)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.setLabelsOnNode(ExceptionTranslatingQueryContext.scala:53)
	at org.neo4j.cypher.internal.runtime.interpreted.UpdateCountingQueryContext.setLabelsOnNode(UpdateCountingQueryContext.scala:88)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.SetLabelsOperation.set(SetOperation.scala:265)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.SetPipe$$anonfun$internalCreateResults$1.apply(SetPipe.scala:32)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.SetPipe$$anonfun$internalCreateResults$1.apply(SetPipe.scala:31)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EmptyResultPipe.internalCreateResults(EmptyResultPipe.scala:29)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:76)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:72)
	at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.BaseExecutionResultBuilderFactory$BaseExecutionWorkflowBuilder.build(DefaultExecutionResultBuilderFactory.scala:61)
	at org.neo4j.cypher.internal.compatibility.InterpretedRuntime$InterpretedExecutionPlan.run(InterpretedRuntime.scala:84)
	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:205)
	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:190)
	at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery.execute(CypherCurrentCompiler.scala:224)
	at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:101)
	at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:85)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachineV1SPI$BoltResultHandleV1.start(TransactionStateMachineV1SPI.java:189)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:506)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:300)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:269)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:110)
	at org.neo4j.bolt.v3.runtime.ReadyState.processRunMessage(ReadyState.java:85)
	at org.neo4j.bolt.v3.runtime.ReadyState.processUnsafe(ReadyState.java:56)
	at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48)
	at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.nextState(BoltStateMachineV1.java:144)
	at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.process(BoltStateMachineV1.java:92)
	at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:89)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:191)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:139)
	at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:171)
	at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:154)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by alexaverbuch** [comment]
@hs750 are you able to provide us with a store that will cause these queries to fail?
i.e., a store that will cause the failure when we run
```
UNWIND [{_id:48, properties:{password:"asdf", authToken:"6a4e0954-3d78-4601-bd7f-0825f9838841", screenName:"asdf", uuid:"96c5e031-8b6c-11e9-9c7e-f8b156dff1f6", username:"asdf"}}] AS row
CREATE (n:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`: row._id})
SET n += row.properties SET n:User;
```
and not cause the failure when we run
```
UNWIND [{_id:1290, properties:{default:true, name:"default", uuid:"96c5e030-8b6c-11e9-9c7e-f8b156dff1f6"}}, {_id:1414, properties:{name:"demo", uuid:"fd460d90-a4c5-11e9-9266-f7d63871dc56"}}] AS row
CREATE (n:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`: row._id})
SET n += row.properties SET n:Context;
```
As you wrote, the issue does not happen if the database is empty, so it's difficult to investigate without such a store.

Thanks

**Comment by sherfert** [comment]
@hs750 We unfortunately still need additional information to be able to act on this bug report. Please reopen the issue with a store or help us otherwise reproduce it.

**Comment by hs750** [comment]
@sherfert @alexaverbuch Sorry for the ultra late response. I have just retested with Neo4j 3.5.6 and 3.5.9 and it is still an issue. [Here is an a database dump](https://github.com/neo4j/neo4j/files/3608997/database.zip) that has the issue (zipped as github wont accept other file types). Running the cypher statement in my first comment will cause the exception. 

The below stack trace is from neo4j 3.5.9 and appears to be the same as that from my original post.
```
2019-09-13 07:58:32.724+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: null, reference 1d61b1f7-74b3-4083-95de-2ab529a3c78d.
2019-09-13 07:58:32.726+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: null, reference 1d61b1f7-74b3-4083-95de-2ab529a3c78d.
java.lang.NullPointerException
	at org.neo4j.values.storable.ValueTuple.hashCode(ValueTuple.java:108)
	at java.util.HashMap.hash(Unknown Source)
	at java.util.HashMap.computeIfAbsent(Unknown Source)
	at org.neo4j.kernel.impl.api.state.TxState.getOrCreateIndexUpdatesForSeek(TxState.java:803)
	at org.neo4j.kernel.impl.api.state.TxState.indexDoUpdateEntry(TxState.java:786)
	at org.neo4j.kernel.impl.newapi.IndexTxStateUpdater.onLabelChange(IndexTxStateUpdater.java:87)
	at org.neo4j.kernel.impl.newapi.Operations.checkConstraintsAndAddLabelToNode(Operations.java:303)
	at org.neo4j.kernel.impl.newapi.Operations.nodeAddLabel(Operations.java:273)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext$$anonfun$setLabelsOnNode$1.apply$mcIII$sp(TransactionBoundQueryContext.scala:81)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext$$anonfun$setLabelsOnNode$1.apply(TransactionBoundQueryContext.scala:80)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext$$anonfun$setLabelsOnNode$1.apply(TransactionBoundQueryContext.scala:80)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
	at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1334)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext.setLabelsOnNode(TransactionBoundQueryContext.scala:80)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$setLabelsOnNode$1.apply$mcI$sp(ExceptionTranslatingQueryContext.scala:54)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$setLabelsOnNode$1.apply(ExceptionTranslatingQueryContext.scala:54)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$setLabelsOnNode$1.apply(ExceptionTranslatingQueryContext.scala:54)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$class.translateException(ExceptionTranslationSupport.scala:33)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.translateException(ExceptionTranslatingQueryContext.scala:42)
	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.setLabelsOnNode(ExceptionTranslatingQueryContext.scala:54)
	at org.neo4j.cypher.internal.runtime.interpreted.UpdateCountingQueryContext.setLabelsOnNode(UpdateCountingQueryContext.scala:88)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.SetLabelsOperation.set(SetOperation.scala:265)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.SetPipe$$anonfun$internalCreateResults$1.apply(SetPipe.scala:32)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.SetPipe$$anonfun$internalCreateResults$1.apply(SetPipe.scala:31)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EmptyResultPipe.internalCreateResults(EmptyResultPipe.scala:29)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:76)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:72)
	at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.BaseExecutionResultBuilderFactory$BaseExecutionWorkflowBuilder.build(DefaultExecutionResultBuilderFactory.scala:61)
	at org.neo4j.cypher.internal.compatibility.InterpretedRuntime$InterpretedExecutionPlan.run(InterpretedRuntime.scala:84)
	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:205)
	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:190)
	at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery.execute(CypherCurrentCompiler.scala:224)
	at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:101)
	at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:85)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachineV1SPI$BoltResultHandleV1.start(TransactionStateMachineV1SPI.java:189)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:506)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:300)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:269)
	at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:110)
	at org.neo4j.bolt.v3.runtime.ReadyState.processRunMessage(ReadyState.java:85)
	at org.neo4j.bolt.v3.runtime.ReadyState.processUnsafe(ReadyState.java:56)
	at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48)
	at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.nextState(BoltStateMachineV1.java:144)
	at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.process(BoltStateMachineV1.java:92)
	at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:89)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:191)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:139)
	at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:171)
	at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:154)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by sherfert** [comment]
Thanks @hs750 . We'll have another look.

**Comment by SecPrez** [comment]
We are hitting the same callstack when we try to insert data into a graph that has a full text search index. Without the index the inserts work, with the full text index we get the exception above.

**Comment by chrisvest** [comment]
This will be fix in the next 3.5 release.
